remote-trait-object/src/forwarder.rs

```diff
@@ -37,24 +39,30 @@ impl ServiceForwarder {
     pub fn new() -> Self {
         Self {
             service_handlers: Default::default(),
+            tokens: RwLock::new({
+                let mut queue = VecDeque::new();
+                for i in 0..100 {
+                    queue.push_back(i)
+                }
+                queue
+            }),
         }
     }
 
-    pub fn add_service(&self, name: ServiceId, service: Box<dyn Dispatch>) {
-        let insert_result = self.service_handlers.write().insert(name.clone(), service);
-        if insert_result.is_some() {
-            panic!("Duplicated service id {}", name);
-        }
+    pub fn add_service(&self, service: Arc<dyn Dispatch>) -> ServiceObjectId {
```

Comment:
How about `add_service_object(&self, service_object: Arc<dyn Dispatch>)`?

Reply:
@majecty I used `register` instead of `add`.